share/revng/test/configuration/revng/scripting.yml

```diff
@@ -0,0 +1,171 @@
+        function_original_name = ""
+        function_idx = 0
+        for idx, function in enumerate(project.model.Functions):
+            if function.CustomName:
+                function_original_name = function.CustomName
+                function_idx = idx
+                break
+        assert function_original_name != ""
+
+        # Get artifact by using the project and function, the result should be the same
+        result1 = project.model.Functions[function_idx].fetch("disassemble")
+        result2 = project.fetch("disassemble", [project.model.Functions[function_idx]])
+        assert ptml.parse(result1) == ptml.parse(result2)
+
+        # Make sure that we get the correct function name in the parse result
+        re_match = re.search(function_original_name, ptml.parse(result1))
+        assert re_match is not None
+
+        # Change the function name, get an artifact and make sure that the 
+        # parsed result contains the new name
+        function_new_name1 = "lol"
+        project.model.Functions[function_idx].CustomName = function_new_name1
+        result1 = project.model.Functions[function_idx].fetch("disassemble")
+        re_match = re.search(function_new_name1, ptml.parse(result1))
+        assert re_match is not None
+
+        # Change function name again and save
+        function_new_name2 = "lolz"
+        project.model.Functions[function_idx].CustomName = function_new_name2
+        project.save()
+
+        # Load the project again and run the analysis again
+        project.import_binary(sys.argv[1], sys.argv[2])
+        project.analyze("revng-initial-auto-analysis")
+        project.analyze("revng-c-initial-auto-analysis")
+
+        # make sure that the second name changes was saved
+        assert project.model.Functions[function_idx].CustomName == function_new_name2
+
+        # Get multiple artifacts
+        result = project.fetch_many({
+            "disassemble": [
+                project.model.Functions[0],
+                project.model.Functions[1]
+            ],
+            "decompile": []
+        })
+        ptml.parse(result["disassemble"])
+        ptml.parse(result["decompile"])
+
+  #
+  # Test python scripting daemon
+  #
+  - type: revng.scripting-daemon
+    from:
+      - type: revng-qa.compiled
+        filter: example-executable-1 and with-debug-info
+    command: |-
+      python scripting-daemon.py "$OUTPUT" "$INPUT"
+    scripts:
+      scripting-daemon.py: |-
+        #!/usr/bin/env python3
+
+        from revng.scripting import DaemonProject
+        from revng import ptml
+        import subprocess
+        from time import sleep
+        import psutil
+        import atexit
+        import sys
+        import re
+
+
+        def exit_handler(pid):
+            process = psutil.Process(pid)
+            for proc in process.children(recursive=True):
+                proc.kill()
+            process.kill()
```

Comment:
Why is this needed? Issuing `SIGINT` to the daemon should stop gracefully

Reply:
I tried to kill it with `process.send_signal`, but it doesn't work. I think is because I run `Popen` with `shell=True`. How come that running [this](https://github.com/revng/revng/blob/develop/share/revng/test/tests/daemon/test.py#L88) doesn't throw an error? If I do the same it complains that it can't find the `revng`, I could add to `$PATH` but it seems wrong. How do the tests find the executable?

Reply:
> I think is because I run `Popen` with `shell=True`. 

Using `shell=True` is very strange, what wasn't working without it?

> How come that running [this](https://github.com/revng/revng/blob/develop/share/revng/test/tests/daemon/test.py#L88) doesn't throw an error? If I do the same it complains that it can't find the `revng`

The script runs with the `revng` component installed and inside the orchestra environment, where the `PATH` is different, if you want to test it you need to do `orc shell ./script.py`

Reply:
I got it working without the `shell=True`. The problem was that I was passing only the `REVNG_DATA_DIR` env and the script couldn't find the `revng` executable, now I pass all the `envs` together with REVNG_DATA_DIR and it's working.